### ISSUE TITLE
Unable to open database file

### ISSUE THREAD

**Issue by ngodfriedt** [Technosoft2000/docker-calibre-web#17]
Hi,

I'm trying to deploy the docker on a linux NAS running OpenMediaVault, but I run into issues I can't seem to resolve. Can you help me figure this out? Here are my logs:

`        ,----,
      ,/   .`|
    ,`   .'  : .--.--.        ,----,        ,-.
  ;    ;     //  /    '.    .'   .' \   ,--/ /|
.'___,/    ,'|  :  /`. /  ,----,'    |,--. :/ |
|    :     | ;  |  |--`   |    :  .  ;:  : ' /
;    |.';  ; |  :  ;_     ;    |.'  / |  '  /
`----'  |  |  \  \    `.  `----'/  ;  '  |  :
    '   :  ;   `----.   \   /  ;  /   |  |   \
    |   |  '   __ \  \  |  ;  /  /-,  '  : |. \
    '   :  |  /  /`--'  / /  /  /.`|  |  | ' \ \
    ;   |.'  '--'.     /./__;      :  '  : |--'
    '---'      `--'---' |   :    .'   ;  |,'
                        ;   | .'      '--'
                        `---'

      PRESENTS ANOTHER AWESOME DOCKER IMAGE

      ~~~~~         Calibre Web       ~~~~~

[INFO] Docker image version: 1.1.8
[INFO] Alpine Linux version: 3.6.0
[INFO] Create group calibre with id 1001
[INFO] Create user calibre with id 1017
[INFO] Current active timezone is CEST
ln: failed to create symbolic link '/etc/localtime': Resource busy
[INFO] Container timezone is changed to: Europe/Amsterdam
[INFO] Change the ownership of /calibre-web (including subfolders) to calibre:calibre
[INFO] Current git version is:
git version 2.13.5
[INFO] Checkout the latest Calibre-Web version ...
[INFO] ... git clone -b master --single-branch https://github.com/janeczku/calibre-web.git /calibre-web/app -v
Cloning into '/calibre-web/app'...
POST git-upload-pack (189 bytes)
[INFO] Autoupdate is active, try to pull the latest sources for Calibre-Web ...
[INFO] ... current git status is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] ... pulling sources
Already up-to-date.
[INFO] ... git status after update is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] kindlegen (Amazon Kindle Generator) will be linked into /calibre-web/app/vendor
[INFO] Creating the vendor directory: /calibre-web/app/vendor
[INFO] Change the ownership of /calibre-web/app/vendor (including subfolders) to calibre:calibre
> create kindlegen link /calibre-web/app/vendor/kindlegen assigned to source /calibre-web/kindlegen/kindlegen
[INFO] Checking permissions of /books
> Output is: 770 calibre 1001 calibre 1017
> Permissions: 770
> Assigned group: calibre
> Assigned group ID: 1001
> Assigned owner: calibre
> Assigned owner ID: 1017
> Using permissions for checks: 0770
> Check if the group calibre has write access at /books
> The group calibre has write access at /books
[INFO] app.db and gdrive.db will be linked into /books
> create app.db link /calibre-web/app/app.db assigned to source /books/app.db
> create gdrive.db link /calibre-web/app/gdrive.db assigned to source /books/gdrive.db
[INFO] Creating directory for temporary directories and files: /tmp
[INFO] Change the ownership of /tmp (including subfolders) to calibre:calibre
[INFO] Launching Calibre-Web ...
Traceback (most recent call last):
  File "/calibre-web/app/cps.py", line 13, in <module>
    from cps import web
  File "/calibre-web/app/cps/web.py", line 36, in <module>
    import ub
  File "/calibre-web/app/cps/ub.py", line 605, in <module>
    Base.metadata.create_all(engine)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3949, in create_all
    tables=tables)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1928, in _run_visitor
    with self._optional_conn_ctx_manager(connection) as conn:
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1921, in _optional_conn_ctx_manager
    with self.contextual_connect() as conn:
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2112, in contextual_connect
    self._wrap_pool_connect(self.pool.connect, None),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2151, in _wrap_pool_connect
    e, dialect, self)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1465, in _handle_dbapi_exception_noconnection
    exc_info
  File "/usr/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect
    return fn()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 387, in connect
    return _ConnectionFairy._checkout(self)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 766, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 516, in checkout
    rec = pool._do_get()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 1229, in _do_get
    return self._create_connection()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 333, in _create_connection
    return _ConnectionRecord(self)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 461, in __init__
    self.__connect(first_connect_check=True)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 651, in __connect
    connection = pool._invoke_creator(self)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 105, in connect
    return dialect.connect(*cargs, **cparams)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 393, in connect
    return self.dbapi.connect(*cargs, **cparams)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file`

I run the docker with similar command like this;

docker create --name=calibre-web --restart=always \
-v /volume1/books/calibre:/books \
-e SET_CONTAINER_TIMEZONE=true \
-e CONTAINER_TIMEZONE=Europe/Amsterdam \
-e PGID=1001 -e PUID=1017 \
-p 8083:8083 \
technosoft2000/calibre-web

File permissions are like this:

-rwxrwx---+   1 calibre restricted 1498112 Oct 23 09:33 metadata.db                                                                        
-rwxrwx---+   1 calibre restricted   15205 Oct 23 09:49 metadata_db_prefs_backup.json

id calibre                                                                                   
uid=1017(calibre) gid=100(users) groups=100(users),1001(restricted)

I hope you can help me out.

Regards,

Niels

**Comment by Technosoft2000** [comment]
Hi @ngodfriedt,

an interesting error is at the last line `sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file`

The permissions on the metadata.db file is okay, but what permissions has the directory `
/volume1/books/calibre` itself? Please check it with `ls -al /volume1/books` you should have something like `drwxrwx---  1 calibre restricted ... calibre`

**Comment by ngodfriedt** [comment]
This is the output outside of the docker:

drwxrwx---+ 99 calibre restricted     4096 Oct 23 09:49 calibre

And maybe this helps. This is the output of your command inside the docker container:
of the dir:
drwxrwx---+  99 calibre calibre 4096 Oct 23 09:49 books
of the file:
-rwxrwx---+   1 calibre calibre 1498112 Oct 23 09:33  metadata.db

I do not have a directory "books" outside the container. My full path of the calibre database file on the disk is this:

/srv/pool/documenten/Ebooks/calibre

**Comment by Technosoft2000** [comment]
Ok the permissions looks for me correct and it seems that this issue has nothing to do with the _metadata.db_

I've digged into the error line `File "/calibre-web/app/cps/ub.py", line 605, in 
Base.metadata.create_all(engine)` and found in the Calibre Web source code at https://github.com/janeczku/calibre-web/blob/master/cps/ub.py that the issue occurs because the calibre-web application has no possibility to open _app.db_

but I see in your log output that the symlink creation was successful `create app.db link /calibre-web/app/app.db assigned to source /books/app.db`

could you please check in your docker container if you see the _app.db_ symlink and check also `/srv/pool/documenten/Ebooks/calibre` if you can see next to the _metadata.db_ an _app.db_ file

thanks

**Comment by ngodfriedt** [comment]
There is no app.db file in the directory.

I will try to look in the container for the symlink, but the container keeps restarting and I have very little time to browse the for files.

**Comment by ngodfriedt** [comment]
The symlink is indeed created in /calibre-web/app/:

lrwxrwxrwx 1 calibre calibre    13 Oct 25 09:24 app.db -> /books/app.db

**Comment by Technosoft2000** [comment]
Could you please create a new empty directory (without any books and metadata.db) on your NAS e.g. _srv/pool/documenten/Ebooks/calibre-test_ with the same owner and permissions as _srv/pool/documenten/Ebooks/calibre_

and then create a new container

```
docker create --name=calibre-test --restart=always 
-v srv/pool/documenten/Ebooks/calibre-test:/books 
-e SET_CONTAINER_TIMEZONE=true 
-e CONTAINER_TIMEZONE=Europe/Amsterdam 
-e PGID=1001 -e PUID=1017 
-p 8083:8083 
technosoft2000/calibre-web:v1.1.9
```

After starting the container you should get the following similar output:
```
[INFO] Docker image version: 1.1.9
[INFO] Alpine Linux version: 3.6.2
[WARNING] A group with id 100 exists already [in use by users] and will be modified.
[WARNING] The group users will be renamed to calibre
[INFO] Create user calibre with id 1029
[INFO] Current active timezone is UTC
Mon Oct 30 19:36:47 CET 2017
[INFO] Container timezone is changed to: Europe/Amsterdam
[INFO] Change the ownership of /calibre-web (including subfolders) to calibre:calibre
[INFO] Current git version is:
git version 2.13.5
[INFO] Checkout the latest Calibre-Web version ...
[INFO] ... git clone -b master --single-branch https://github.com/janeczku/calibre-web.git /calibre-web/app -v
Cloning into '/calibre-web/app'...
POST git-upload-pack (189 bytes)
[INFO] Autoupdate is active, try to pull the latest sources for Calibre-Web ...
[INFO] ... current git status is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] ... pulling sources
Already up-to-date.
[INFO] ... git status after update is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] kindlegen (Amazon Kindle Generator) will be linked into /calibre-web/app/vendor
[INFO] Creating the vendor directory: /calibre-web/app/vendor
[INFO] Change the ownership of /calibre-web/app/vendor (including subfolders) to calibre:calibre
> create kindlegen link /calibre-web/app/vendor/kindlegen assigned to source /calibre-web/kindlegen/kindlegen
[INFO] Checking permissions of /books
> Output is: 770 calibre 100 UNKNOWN 1026
> Permissions: 770
> Assigned group: calibre
> Assigned group ID: 100
> Assigned owner: UNKNOWN
> Assigned owner ID: 1026
> Using permissions for checks: 0770
> Check if the group calibre has write access at /books
> The group calibre has write access at /books
[INFO] app.db and gdrive.db will be linked into /books
> create app.db link /calibre-web/app/app.db assigned to source /books/app.db
> create gdrive.db link /calibre-web/app/gdrive.db assigned to source /books/gdrive.db
[WARNING] The mapped volume for /books doesn't contain a Calibre database file 'metadata.db'
> Due this an inital Calibre database file 'metadata.db' and 'metadata_db_prefs_backup.json' will be copied to /books
[INFO] Creating directory for temporary directories and files: /tmp
[INFO] Change the ownership of /tmp (including subfolders) to calibre:calibre
[INFO] Launching Calibre-Web ...
```

and your directory _srv/pool/documenten/Ebooks/calibre-test_ should have after startup the following files:
 * app.db
 * gdrive.db
 * metadata.db
 * metadata_db_prefs_backup.json

please provide then again your log output

**Comment by ngodfriedt** [comment]
The test directory now only contains these files:

-rw-r-----+ 1 calibre restricted 348160 Oct 31 09:07 metadata.db                                    
-rw-r-----+ 1 calibre restricted  12900 Oct 31 09:07 metadata_db_prefs_backup.json

and this is the log output:

        ,----,
      ,/   .`|
    ,`   .'  : .--.--.        ,----,        ,-.
  ;    ;     //  /    '.    .'   .' \   ,--/ /|
.'___,/    ,'|  :  /`. /  ,----,'    |,--. :/ |
|    :     | ;  |  |--`   |    :  .  ;:  : ' /
;    |.';  ; |  :  ;_     ;    |.'  / |  '  /
`----'  |  |  \  \    `.  `----'/  ;  '  |  :
    '   :  ;   `----.   \   /  ;  /   |  |   \
    |   |  '   __ \  \  |  ;  /  /-,  '  : |. \
    '   :  |  /  /`--'  / /  /  /.`|  |  | ' \ \
    ;   |.'  '--'.     /./__;      :  '  : |--'
    '---'      `--'---' |   :    .'   ;  |,'
                        ;   | .'      '--'
                        `---'

      PRESENTS ANOTHER AWESOME DOCKER IMAGE

      ~~~~~         Calibre Web       ~~~~~

[INFO] Docker image version: 1.1.8
[INFO] Alpine Linux version: 3.6.0
[INFO] Create group calibre with id 1001
[INFO] Create user calibre with id 1017
[INFO] Current active timezone is CET
ln: failed to create symbolic link '/etc/localtime': Resource busy
[INFO] Container timezone is changed to: Europe/Amsterdam
[INFO] Change the ownership of /calibre-web (including subfolders) to calibre:calibre
[INFO] Current git version is:
git version 2.13.5
[INFO] Checkout the latest Calibre-Web version ...
[INFO] ... git clone -b master --single-branch https://github.com/janeczku/calibre-web.git /calibre-web/app -v
Cloning into '/calibre-web/app'...
POST git-upload-pack (189 bytes)
error: RPC failed; curl 56 LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 104
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
[INFO] Autoupdate is active, try to pull the latest sources for Calibre-Web ...
[INFO] ... current git status is
/init/checkout.sh: line 21: cd: /calibre-web/app: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
fatal: Unable to read current working directory: No such file or directory
[INFO] ... pulling sources
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
fatal: Unable to read current working directory: No such file or directory
[INFO] ... git status after update is
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
fatal: Unable to read current working directory: No such file or directory
[INFO] kindlegen (Amazon Kindle Generator) will be linked into /calibre-web/app/vendor
[INFO] Creating the vendor directory: /calibre-web/app/vendor
[INFO] Change the ownership of /calibre-web/app/vendor (including subfolders) to calibre:calibre
> create kindlegen link /calibre-web/app/vendor/kindlegen assigned to source /calibre-web/kindlegen/kindlegen
[INFO] Checking permissions of /books
> Output is: 770 calibre 1001 calibre 1017
> Permissions: 770
> Assigned group: calibre
> Assigned group ID: 1001
> Assigned owner: calibre
> Assigned owner ID: 1017
> Using permissions for checks: 0770
> Check if the group calibre has write access at /books
> The group calibre has write access at /books
[INFO] app.db and gdrive.db will be linked into /books
> create app.db link /calibre-web/app/app.db assigned to source /books/app.db
> create gdrive.db link /calibre-web/app/gdrive.db assigned to source /books/gdrive.db
[WARNING] The mapped volume for /books doesn't contain a Calibre database file 'metadata.db'
> Due this an inital Calibre database file 'metadata.db' and 'metadata_db_prefs_backup.json' will be copied to /books
[INFO] Creating directory for temporary directories and files: /tmp
[INFO] Change the ownership of /tmp (including subfolders) to calibre:calibre
[INFO] Launching Calibre-Web ...
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
/usr/bin/python: can't open file '/calibre-web/app/cps.py': [Errno 2] No such file or directory

**Comment by Technosoft2000** [comment]
I see you're not using the latest image from yesterday _Docker image version: 1.1.9_ because you've in your log `Docker image version: 1.1.8` 

and it is interessting to see that you get issues which I can't reproduce at all, like:
 * `ln: failed to create symbolic link '/etc/localtime': Resource busy`
 * `error: RPC failed; curl 56 LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 104`
 * `/init/checkout.sh: line 21: cd: /calibre-web/app: No such file or directory`

so please remove your container and the image an pull the version **technosoft2000/calibre-web:v1.1.9** for further tests.

I download now OpenMediaVault v3.0.86 from https://sourceforge.net/projects/openmediavault/files/3.0.86/ and install it on a Virtual Machine (VirtualBox) to check if I can reproduce the issues above.

**Comment by Technosoft2000** [comment]
1. installed openmediavault_3.0.86-amd64.iso

2. created via openmediavault web gui an user 'docker'

3. on commandline i've checked if the user exists

```
   # getent passwd docker
   docker:x:1000:100::/home/docker:/bin/bash

   # id docker
   uid=1000(docker) gid=100(users) groups=100(users)
```
   
4. installed via openmediavault web gui the plugin **openmediavault-omvextrasorg_latest_all3.deb**
   from http://omv-extras.org
   
5. activated via openmediavault web gui at **OMV-Extras** the repo **Docker CE**

6. installed via openmediavault web gui the plugin **openmediavault-docker-gui 3.1.10**

7. on commandline i've checked if `docker` is available

```
   # docker version
   
   Client:
    Version:      17.09.0-ce
	API version:  1.32
	Go version:   go1.8.3
	Git commit:   afdb6d4
	Built:        Tue Sep 26 22:40:46 2017
	OS/Arch:      linux/amd64
	
   Server:
    Version:      17.09.0-ce
	API version:  1.32 (minimum version 1.12)
	Go version:   go1.8.3
	Git commit:   afdb6d4
	Built:        Tue Sep 26 22:39:27 2017
	OS/Arch:      linux/amd64
	Experimental: false
```
	
8. activated via openmediavault web gui the Docker overview

9. created with commandline the directory _/volume1/books/calibre_
   with `chown -R root:users /volume1/` and `chmod -R 770 /volume1/`

10. pulled manually the docker image on commandline

```
   # docker pull technosoft2000/calibre-web:v1.1.9
```

11. created the docker container on commandline

```
   # docker create --name=calibre-web --restart=always \
    -v /volume1/books/calibre:/books \
    -e SET_CONTAINER_TIMEZONE=true \
    -e CONTAINER_TIMEZONE=Europe/Amsterdam \
    -e PGID=100 -e PUID=1000 \
    -p 8082:8083 \
    technosoft2000/calibre-web:v1.1.9
```
	
12. check if the container is okay

```	
   # docker ps -a
   this shows that the container was successfully created
```
   
13. start the container

```
   # docker start calibre-web
```
   
14. via openmediavault web gui I've checked the running container and looked at the log and got the following


```
       ,----,
      ,/   .`|
    ,`   .'  : .--.--.        ,----,        ,-.
  ;    ;     //  /    '.    .'   .' \   ,--/ /|
.'___,/    ,'|  :  /`. /  ,----,'    |,--. :/ |
|    :     | ;  |  |--`   |    :  .  ;:  : ' /
;    |.';  ; |  :  ;_     ;    |.'  / |  '  /
`----'  |  |  \  \    `.  `----'/  ;  '  |  :
    '   :  ;   `----.   \   /  ;  /   |  |   \
    |   |  '   __ \  \  |  ;  /  /-,  '  : |. \
    '   :  |  /  /`--'  / /  /  /.`|  |  | ' \ \
    ;   |.'  '--'.     /./__;      :  '  : |--'
    '---'      `--'---' |   :    .'   ;  |,'
                        ;   | .'      '--'
                        `---'

      PRESENTS ANOTHER AWESOME DOCKER IMAGE

      ~~~~~         Calibre Web       ~~~~~

[INFO] Docker image version: 1.1.9
[INFO] Alpine Linux version: 3.6.2
[WARNING] A group with id 100 exists already [in use by users] and will be modified.
[WARNING] The group users will be renamed to calibre
[INFO] Create user calibre with id 1000
[INFO] Current active timezone is UTC
Tue Oct 31 10:54:19 CET 2017
[INFO] Container timezone is changed to: Europe/Amsterdam
[INFO] Change the ownership of /calibre-web (including subfolders) to calibre:calibre
[INFO] Current git version is:
git version 2.13.5
[INFO] Checkout the latest Calibre-Web version ...
[INFO] ... git clone -b master --single-branch https://github.com/janeczku/calibre-web.git /calibre-web/app -v
Cloning into '/calibre-web/app'...
POST git-upload-pack (189 bytes)
[INFO] Autoupdate is active, try to pull the latest sources for Calibre-Web ...
[INFO] ... current git status is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] ... pulling sources
Already up-to-date.
[INFO] ... git status after update is
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
9182d8912816c27e67e452334df370c8fdc89ba4
[INFO] kindlegen (Amazon Kindle Generator) will be linked into /calibre-web/app/vendor
[INFO] Creating the vendor directory: /calibre-web/app/vendor
[INFO] Change the ownership of /calibre-web/app/vendor (including subfolders) to calibre:calibre
> create kindlegen link /calibre-web/app/vendor/kindlegen assigned to source /calibre-web/kindlegen/kindlegen
[INFO] Checking permissions of /books
> Output is: 770 calibre 100 root 0
> Permissions: 770
> Assigned group: calibre
> Assigned group ID: 100
> Assigned owner: root
> Assigned owner ID: 0
> Using permissions for checks: 0770
> Check if the group calibre has write access at /books
> The group calibre has write access at /books
[INFO] app.db and gdrive.db will be linked into /books
> create app.db link /calibre-web/app/app.db assigned to source /books/app.db
> create gdrive.db link /calibre-web/app/gdrive.db assigned to source /books/gdrive.db
[WARNING] The mapped volume for /books doesn't contain a Calibre database file 'metadata.db'
> Due this an inital Calibre database file 'metadata.db' and 'metadata_db_prefs_backup.json' will be copied to /books
[INFO] Creating directory for temporary directories and files: /tmp
[INFO] Change the ownership of /tmp (including subfolders) to calibre:calibre
[INFO] Launching Calibre-Web ...   
```
   
So it is up and running without any issue.

15. I've checked the directory _/volume1/books/calibre_ and it contains
   * app.db
   * gdrive.db
   * metadata.db
   * metadata_db_prefs_backup.json
   
16. I've checked if I can access via `http://<HOSTNAME>:<PORT>`
    e.g. in my case _http://localhost:8082_ if can reach the Calibre Web user interface
	and it works without any issue
	
17. configured the Calibre Web user interface and successfully uploaded one e-book (epub) 

**So what could be the problem on your side:**
 - the pulled Calibre Web image is not okay on your local disk, so remove it and get the actual **technosoft2000/calibre-web:v1.1.9** version
 - or the Docker installation is maybe not okay on your NAS, try to uninstall and reinstall Docker and be aware that you use the newer **Docker CE repo** via the **OpenMediaVault Extra** plugin

**Comment by ngodfriedt** [comment]
Ok, I've made progress. I've made a new share for my ebooks and calibre: /srv/pool/ebooks/calibre

I've set a permissions as prescribed, but have left the directory empty at first.

Then I created a new container et voilá calibre-web is running with an empty database.

Can I now just copy my old database and books to the new share? Would that work? I will stop the docker container of course and set the permissions right again.
